src/types/dict.rs

```diff
@@ -653,7 +653,18 @@ impl<'py> IntoIterator for Bound<'py, PyDict> {
 pub trait IntoPyDict {
     /// Converts self into a `PyDict` object pointer. Whether pointer owned or borrowed
     /// depends on implementation.
+    #[cfg_attr(
+        not(feature = "gil-refs"),
+        deprecated(
+            since = "0.21.0",
+            note = "`IntoPyDict::into_py_dict` will be replaced by `IntoPyDict::into_py_dict_bound` in a future PyO3 version"
+        )
+    )]
     fn into_py_dict(self, py: Python<'_>) -> &PyDict;
```

Comment:
Should we move the default implementation to be part of the trait definition? That way, just in case there are any implementors downstream (unlikely), they just need to implement the new `_bound` method.

Reply:
That's a good idea! I'll add that.

Reply:
Ah, if we want to do that, than we have to require `Sized` for either the full trait or the method with the default implementation. I thinks that's technically a breaking change. I assume that its very unlikely to affect anyone, so should we do it anyway?

Reply:
Hmm curious. I guess because in a trait where other methods might take `&self`, then it might be possible to implement the whole trait for `!Sized` types and leave this method defaulted. But then the method here would be invalid to call without this `where Self: Sized` bound to restrict it.

Anyway, I tested around and adding `where Self: Sized` here doesn't seem to be a breaking change to downstream implementations of the trait; they don't need to add it. It shouldn't break any callers either because all those types should be `Sized` already, as you say.

(I checked even on MSRV.)

So I _think_ it's fine to add the bound and move the default onto the method.

Reply:
(I'd probably put `where Self: Sized` on the method, so that we remove it again when we delete the method and tidy up.)

Reply:
Ah I see you already put it on the trait. That actually seems totally reasonable; I just checked and `Into` also has such a `Sized` bound.